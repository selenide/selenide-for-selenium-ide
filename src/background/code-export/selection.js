import {codeExport as exporter} from '@seleniumhq/side-utils'

const emitters = {
  id: emitId,
  value: emitValue,
  label: emitLabel,
  index: emitIndex,
}

export function emit(location) {
  return exporter.emit.selection(location, emitters)
}

export default {
  emit,
}

function emitId(id) {
  return Promise.resolve(`$(By.cssSelector("#${id}")).click();`)
}

function emitValue(value) {
  return Promise.resolve(`selectOptionByValue("${value}");`)
}

function emitLabel(label) {
  return Promise.resolve(`selectOption("${label}");`)
}

function emitIndex(index) {
  return Promise.resolve(`selectOption(${index});`)
}
