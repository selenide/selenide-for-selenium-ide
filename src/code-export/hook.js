import {codeExport as exporter} from '@seleniumhq/side-utils';

const emitters = {
  afterAll,
  afterEach,
  beforeAll,
  beforeEach,
  declareDependencies,
  declareMethods: empty,
  declareVariables,
  inEachBegin: empty,
  inEachEnd: empty
};

function generate(hookName) {
  return new exporter.hook(emitters[hookName]());
}

export function generateHooks() {
  let result = {};
  Object.keys(emitters).forEach(hookName => {
    result[hookName] = generate(hookName);
  });
  return result;
}

function afterAll() {
  const params = {
    startingSyntax: {
      commands: [
        {level: 0, statement: '@AfterAll'},
        {level: 0, statement: 'public static void finalTearDown() {'}
      ]
    },
    endingSyntax: {
      commands: [{level: 0, statement: '}'}]
    },
    registrationLevel: 1
  };
  return params;
}

function afterEach() {
  const params = {
    startingSyntax: {
      commands: [
        {level: 0, statement: '@AfterEach'},
        {level: 0, statement: 'public void tearDown() {'}
      ]
    },
    endingSyntax: {
      commands: [{level: 0, statement: '}'}]
    }
  };
  return params;
}

function beforeAll() {
  return {
    startingSyntax: {
      commands: [
        {level: 0, statement: '@BeforeAll'},
        {level: 0, statement: 'public static void initialSetUp() {'}
      ]
    },
    endingSyntax: {
      commands: [{level: 0, statement: '}'}]
    },
    registrationLevel: 1
  };
}

function beforeEach() {
  return {
    startingSyntax: ({browserName, gridUrl} = {}) => ({
      commands: [
        {level: 0, statement: '@BeforeEach'},
        {
          level: 0,
          statement: `public void setUp() {`
        },
        {
          level: 1,
          statement: gridUrl
            ? `Configuration.browser = "${browserName ? browserName.toLowerCase() : 'chrome'}";\n    Configuration.remote = "${gridUrl}";`
            : `Configuration.browser = "${browserName ? browserName.toLowerCase() : 'chrome'}";`
        },
        {level: 1, statement: 'vars = new HashMap<>();'}
      ]
    }),
    endingSyntax: {
      commands: [{level: 0, statement: '}'}]
    }
  };
}

function declareDependencies() {
  const params = {
    startingSyntax: {
      commands: [
        {level: 0, statement: 'import org.junit.jupiter.api.Test;'},
        {level: 0, statement: 'import org.junit.jupiter.api.BeforeAll;'},
        {level: 0, statement: 'import org.junit.jupiter.api.BeforeEach;'},
        {level: 0, statement: 'import org.junit.jupiter.api.AfterAll;'},
        {level: 0, statement: 'import org.junit.jupiter.api.AfterEach;'},
        {level: 0, statement: 'import com.codeborne.selenide.Configuration;'},
        {level: 0, statement: 'import com.codeborne.selenide.WebDriverRunner;'},
        {level: 0, statement: 'import static com.codeborne.selenide.Selectors.*;'},
        {level: 0, statement: 'import static com.codeborne.selenide.Condition.*;'},
        {level: 0, statement: 'import static com.codeborne.selenide.Selenide.*;'},
        {level: 0, statement: 'import static org.junit.jupiter.api.Assertions.*;'},
        {level: 0, statement: 'import org.openqa.selenium.Dimension;'},
        {level: 0, statement: 'import org.openqa.selenium.Keys;'},
        {level: 0, statement: 'import java.util.*;'}
      ]
    }
  };
  return params;
}

function declareVariables() {
  const params = {
    startingSyntax: {
      commands: [
        {
          level: 0,
          statement: 'private Map<String, Object> vars;'
        }
      ]
    }
  };
  return params;
}

function empty() {
  return {};
}
