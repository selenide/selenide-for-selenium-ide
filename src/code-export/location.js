"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.emit = emit;
exports.default = void 0;

let _sideUtils = require("@seleniumhq/side-utils");

const emitters = {
  id: emitId,
  name: emitName,
  link: emitLink,
  linkText: emitLink,
  partialLinkText: emitPartialLinkText,
  css: emitCss,
  xpath: emitXpath
};

function emit(location) {
  return _sideUtils.codeExport.emit.location(location, emitters);
}

let _default = {
  emit
};
exports.default = _default;

function emitId(selector) {
  return Promise.resolve(`"#${selector}"`);
}

function emitName(selector) {
  return Promise.resolve(`byName("${selector}")`);
}

function emitLink(selector) {
  return Promise.resolve(`byLinkText("${selector}")`);
}

function emitPartialLinkText(selector) {
  return Promise.resolve(`byPartialLinkText("${selector}")`);
}

function emitCss(selector) {
  return Promise.resolve(`"${selector}"`);
}

function emitXpath(selector) {
  return Promise.resolve(`byXpath("${selector}")`);
}
