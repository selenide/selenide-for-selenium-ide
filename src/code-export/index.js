import {codeExport as exporter} from '@seleniumhq/side-utils';
import emitter from './command';
import location from './location';
import {generateHooks} from './hook';

// Define language options
export const displayName = 'Java Selenide';

export let opts = {};
opts.emitter = emitter;
opts.hooks = generateHooks();
opts.fileExtension = '.java';
opts.commandPrefixPadding = '  ';
opts.terminatingKeyword = '}';
opts.commentPrefix = '//';
opts.generateMethodDeclaration = generateMethodDeclaration;

// Create generators for dynamic string creation of primary entities (e.g., filename, methods, test, and suite)
function generateTestDeclaration(name) {
  return `@Test\npublic void ${exporter.parsers.uncapitalize(
    exporter.parsers.sanitizeName(name)
  )}() {`;
}

function generateMethodDeclaration(name) {
  return `public void ${exporter.parsers.sanitizeName(name)}() {`;
}

function generateSuiteDeclaration(name) {
  return `public class ${exporter.parsers.capitalize(
    exporter.parsers.sanitizeName(name)
  )}Test {`;
}

function generateFilename(name) {
  return `${exporter.parsers.capitalize(
    exporter.parsers.sanitizeName(name)
  )}Test${opts.fileExtension}`;
}

// Emit an individual test, wrapped in a suite (using the test name as the suite name)
export async function emitTest({
  baseUrl,
  test,
  tests,
  project,
  enableOriginTracing,
  beforeEachOptions,
  enableDescriptionAsComment
}) {
  global.baseUrl = baseUrl;
  const testDeclaration = generateTestDeclaration(test.name);
  const result = await exporter.emit.test(test, tests, {
    ...opts,
    testDeclaration,
    enableOriginTracing,
    enableDescriptionAsComment,
    project
  });
  const suiteName = test.name;
  const suiteDeclaration = generateSuiteDeclaration(suiteName);
  const _suite = await exporter.emit.suite(result, tests, {
    ...opts,
    suiteDeclaration,
    suiteName,
    project,
    beforeEachOptions
  });
  return {
    filename: generateFilename(test.name),
    body: exporter.emit.orderedSuite(_suite)
  };
}

// Emit a suite with all of its tests
export async function emitSuite({
  baseUrl,
  suite,
  tests,
  project,
  enableOriginTracing,
  beforeEachOptions,
  enableDescriptionAsComment
}) {
  global.baseUrl = baseUrl;
  const result = await exporter.emit.testsFromSuite(tests, suite, opts, {
    enableOriginTracing,
    enableDescriptionAsComment,
    generateTestDeclaration,
    project
  });
  const suiteDeclaration = generateSuiteDeclaration(suite.name);
  const _suite = await exporter.emit.suite(result, tests, {
    ...opts,
    suiteDeclaration,
    suite,
    project,
    beforeEachOptions
  });
  return {
    filename: generateFilename(suite.name),
    body: exporter.emit.orderedSuite(_suite)
  };
}

export default {
  emit: {
    test: emitTest,
    suite: emitSuite,
    locator: location.emit
  },
  register: {
    command: emitter.register,
    dependency: opts.hooks.declareDependencies.register,
    inEachBegin: opts.hooks.inEachBegin.register,
    inEachEnd: opts.hooks.inEachEnd.register
  }
};
