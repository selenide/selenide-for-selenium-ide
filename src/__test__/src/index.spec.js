import fs from 'fs';
import path from 'path';
import {emitSuite, emitTest} from '../../code-export';
import {project as projectProcessor} from '@seleniumhq/side-utils';

function readFile(filename) {
  return JSON.parse(
    fs.readFileSync(
      path.join(
        __dirname,
        '..',
        'test-files',
        filename
      )
    )
  );
}

describe('Code Export Java Selenide', () => {
  it('should export a test', async () => {
    const project = readFile('single-test.side');
    const results = await emitTest({
      baseUrl: project.url,
      test: project.tests[0],
      tests: project.tests
    });
    expect(results.body).toBeDefined();
    expect(results.body).toMatchSnapshot();
  });
  it('should export a test with a grid configuration', async () => {
    const project = readFile('single-test.side');
    const results = await emitTest({
      baseUrl: project.url,
      test: project.tests[0],
      tests: project.tests,
      beforeEachOptions: {
        browserName: 'Firefox',
        gridUrl: 'http://localhost:4444/wd/hub'
      }
    });
    expect(results.body).toBeDefined();
    expect(results.body).toMatchSnapshot();
  });
  it('should export a suite', async () => {
    const project = normalizeProject(readFile('single-suite.side'));
    const results = await emitSuite({
      baseUrl: project.url,
      suite: project.suites[0],
      tests: project.tests
    });
    expect(results.body).toBeDefined();
    expect(results.body).toMatchSnapshot();
  });
  it('should export a test with a reused test method', async () => {
    const project = normalizeProject(readFile('test-case-reuse.side'));
    const results = await emitTest({
      baseUrl: project.url,
      test: project.tests[1],
      tests: project.tests
    });
    expect(results.body).toBeDefined();
    expect(results.body).toMatchSnapshot();
  });
  it('should export a test with commands that open a new window', async () => {
    const project = normalizeProject(readFile('select-window.side'));
    const results = await emitTest({
      baseUrl: project.url,
      test: project.tests[0],
      tests: project.tests
    });
    expect(results.body).toBeDefined();
    expect(results.body).toMatchSnapshot();
  });
  it('should export a suite with a reused test method', async () => {
    const project = normalizeProject(readFile('test-case-reuse.side'));
    const results = await emitSuite({
      baseUrl: project.url,
      suite: project.suites[0],
      tests: project.tests
    });
    expect(results.body).toBeDefined();
    expect(results.body).toMatchSnapshot();
  });
  it('should export a suite that uses control flow commands', async () => {
    const project = normalizeProject(readFile('control-flow-suite.side'));
    const results = await emitSuite({
      baseUrl: project.url,
      suite: project.suites[0],
      tests: project.tests
    });
    expect(results.body).toBeDefined();
    expect(results.body).toMatchSnapshot();
  });
  it('should export a suite with commands that open a new window inside of a reused test method', async () => {
    const project = normalizeProject(readFile('nested-select-window.side'));
    const results = await emitSuite({
      baseUrl: project.url,
      suite: project.suites[0],
      tests: project.tests
    });
    expect(results.body).toBeDefined();
    expect(results.body).toMatchSnapshot();
  });
  it('should export a suite with just one new window util method when there are multiple commands that open a new window', async () => {
    const project = normalizeProject(readFile('nested-select-window-v2.side'));
    const results = await emitSuite({
      baseUrl: project.url,
      suite: project.suites[0],
      tests: project.tests
    });
    expect(results.body).toBeDefined();
    expect(results.body).toMatchSnapshot();
  });
});

function normalizeProject(project) {
  let _project = {...project};
  _project.suites.forEach(suite => {
    projectProcessor.normalizeTestsInSuite({suite, tests: _project.tests});
  });
  return _project;
}
