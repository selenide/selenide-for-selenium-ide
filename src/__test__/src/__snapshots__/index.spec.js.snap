// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Code Export Java Selenide should export a suite 1`] = `
"// Generated by Selenium IDE
import org.junit.jupiter.api.Test;
import org.junit.jupiter.api.BeforeAll;
import org.junit.jupiter.api.BeforeEach;
import org.junit.jupiter.api.AfterAll;
import org.junit.jupiter.api.AfterEach;
import com.codeborne.selenide.Configuration;
import com.codeborne.selenide.Selectors;
import com.codeborne.selenide.Selenide;
import com.codeborne.selenide.WebDriverRunner;
import static com.codeborne.selenide.Condition.*;
import static com.codeborne.selenide.Selenide.*;
import static org.junit.jupiter.api.Assertions.*;
import org.openqa.selenium.Dimension;
import org.openqa.selenium.Keys;
import java.util.*;
public class LoginTest {
  private Map<String, Object> vars;
  @BeforeEach
  public void setUp() {
    Configuration.browser = \\"chrome\\";
    vars = new HashMap<>();
  }
  @AfterEach
  public void tearDown() {
  }
  @Test
  public void validcredentials() {
    Selenide.open(\\"http://the-internet.herokuapp.com/login\\");
    WebDriverRunner.getWebDriver().manage().window().setSize(new Dimension(1440, 1177));
    $(\\"#username\\").val(\\"tomsmith\\");
    $(\\"#password\\").val(\\"SuperSecretPassword!\\");
    $(\\"#login button\\").click();
    $(\\".flash.success\\").shouldHave(text(\\"You logged into a secure area!\\\\\\\\\\\\\\\\n×\\"));
  }
  @Test
  public void invalidcredentials() {
    Selenide.open(\\"http://the-internet.herokuapp.com/login\\");
    WebDriverRunner.getWebDriver().manage().window().setSize(new Dimension(1440, 1177));
    $(\\"#username\\").val(\\"blah\\");
    $(\\"#password\\").val(\\"blah\\");
    $(\\"#login button\\").click();
    $(\\"#flash\\").shouldHave(text(\\"Your username is invalid!\\\\\\\\\\\\\\\\n×\\"));
  }
}
"
`;

exports[`Code Export Java Selenide should export a suite that uses control flow commands 1`] = `
"// Generated by Selenium IDE
import org.junit.jupiter.api.Test;
import org.junit.jupiter.api.BeforeAll;
import org.junit.jupiter.api.BeforeEach;
import org.junit.jupiter.api.AfterAll;
import org.junit.jupiter.api.AfterEach;
import com.codeborne.selenide.Configuration;
import com.codeborne.selenide.Selectors;
import com.codeborne.selenide.Selenide;
import com.codeborne.selenide.WebDriverRunner;
import static com.codeborne.selenide.Condition.*;
import static com.codeborne.selenide.Selenide.*;
import static org.junit.jupiter.api.Assertions.*;
import org.openqa.selenium.Dimension;
import org.openqa.selenium.Keys;
import java.util.*;
public class ControlflowTest {
  private Map<String, Object> vars;
  @BeforeEach
  public void setUp() {
    Configuration.browser = \\"chrome\\";
    vars = new HashMap<>();
  }
  @AfterEach
  public void tearDown() {
  }
  @Test
  public void controlflowif() {
    vars.put(\\"myVar\\", Selenide.executeJavaScript(\\"return 'a'\\"));
    if ((Boolean) Selenide.executeJavaScript(\\"return (arguments[0] === 'a')\\", vars.get(\\"myVar\\"))) {
      vars.put(\\"output\\", Selenide.executeJavaScript(\\"return 'a'\\"));
    } else if ((Boolean) Selenide.executeJavaScript(\\"return (arguments[0] === 'b')\\", vars.get(\\"myVar\\"))) {
      vars.put(\\"output\\", Selenide.executeJavaScript(\\"return 'b'\\"));
    } else {
      vars.put(\\"output\\", Selenide.executeJavaScript(\\"return 'c'\\"));
    }
    assertEquals(vars.get(\\"output\\").toString(), \\"a\\");
  }
  @Test
  public void controlflowelseif() {
    vars.put(\\"myVar\\", Selenide.executeJavaScript(\\"return 'b'\\"));
    if ((Boolean) Selenide.executeJavaScript(\\"return (arguments[0] === 'a')\\", vars.get(\\"myVar\\"))) {
      vars.put(\\"output\\", Selenide.executeJavaScript(\\"return 'a'\\"));
    } else if ((Boolean) Selenide.executeJavaScript(\\"return (arguments[0] === 'b')\\", vars.get(\\"myVar\\"))) {
      vars.put(\\"output\\", Selenide.executeJavaScript(\\"return 'b'\\"));
    } else {
      vars.put(\\"output\\", Selenide.executeJavaScript(\\"return 'c'\\"));
    }
    assertEquals(vars.get(\\"output\\").toString(), \\"b\\");
  }
  @Test
  public void controlflowelse() {
    vars.put(\\"myVar\\", Selenide.executeJavaScript(\\"return 'c'\\"));
    if ((Boolean) Selenide.executeJavaScript(\\"return (arguments[0] === 'a')\\", vars.get(\\"myVar\\"))) {
      vars.put(\\"output\\", Selenide.executeJavaScript(\\"return 'a'\\"));
    } else if ((Boolean) Selenide.executeJavaScript(\\"return (arguments[0] === 'b')\\", vars.get(\\"myVar\\"))) {
      vars.put(\\"output\\", Selenide.executeJavaScript(\\"return 'b'\\"));
    } else {
      vars.put(\\"output\\", Selenide.executeJavaScript(\\"return 'c'\\"));
    }
    assertEquals(vars.get(\\"output\\").toString(), \\"c\\");
  }
  @Test
  public void controlflowdo() {
    vars.put(\\"check\\", Selenide.executeJavaScript(\\"return 1\\"));
    do {
      vars.put(\\"check\\", Selenide.executeJavaScript(\\"return arguments[0] + 1\\", vars.get(\\"check\\")));
    } while ((Boolean) Selenide.executeJavaScript(\\"return (arguments[0] < 3)\\", vars.get(\\"check\\")));
    assertEquals(vars.get(\\"check\\").toString(), \\"3\\");
  }
  @Test
  public void controlflowtimes() {
    vars.put(\\"check\\", Selenide.executeJavaScript(\\"return 1\\"));
    Integer times = 2;
    for(int i = 0; i < times; i++) {
      vars.put(\\"check\\", Selenide.executeJavaScript(\\"return arguments[0] + 1\\", vars.get(\\"check\\")));
    }
    assertEquals(vars.get(\\"check\\").toString(), \\"3\\");
  }
  @Test
  public void controlflowwhile() {
    vars.put(\\"check\\", Selenide.executeJavaScript(\\"return 1\\"));
    while ((Boolean) Selenide.executeJavaScript(\\"return (arguments[0] < 3)\\", vars.get(\\"check\\"))) {
      vars.put(\\"check\\", Selenide.executeJavaScript(\\"return arguments[0] + 1\\", vars.get(\\"check\\")));
    }
    assertEquals(vars.get(\\"check\\").toString(), \\"3\\");
  }
}
"
`;

exports[`Code Export Java Selenide should export a suite with a reused test method 1`] = `
"// Generated by Selenium IDE
import org.junit.jupiter.api.Test;
import org.junit.jupiter.api.BeforeAll;
import org.junit.jupiter.api.BeforeEach;
import org.junit.jupiter.api.AfterAll;
import org.junit.jupiter.api.AfterEach;
import com.codeborne.selenide.Configuration;
import com.codeborne.selenide.Selectors;
import com.codeborne.selenide.Selenide;
import com.codeborne.selenide.WebDriverRunner;
import static com.codeborne.selenide.Condition.*;
import static com.codeborne.selenide.Selenide.*;
import static org.junit.jupiter.api.Assertions.*;
import org.openqa.selenium.Dimension;
import org.openqa.selenium.Keys;
import java.util.*;
public class LoginTest {
  private Map<String, Object> vars;
  @BeforeEach
  public void setUp() {
    Configuration.browser = \\"chrome\\";
    vars = new HashMap<>();
  }
  @AfterEach
  public void tearDown() {
  }
  public void login() {
    Selenide.open(\\"https://the-internet.herokuapp.com/\\");
    $(Selectors.byLinkText(\\"Form Authentication\\")).click();
    $(\\"#username\\").val(vars.get(\\"username\\").toString());
    $(\\"#password\\").val(vars.get(\\"password\\").toString());
    $(\\".fa\\").click();
  }
  @Test
  public void invalidcredentials() {
    vars.put(\\"username\\", \\"blah\\");
    vars.put(\\"password\\", \\"blah\\");
    login();
  }
  @Test
  public void validcredentials() {
    vars.put(\\"username\\", \\"tomsmith\\");
    vars.put(\\"password\\", \\"SuperSecretPassword!\\");
    login();
  }
}
"
`;

exports[`Code Export Java Selenide should export a suite with commands that open a new window inside of a reused test method 1`] = `
"// Generated by Selenium IDE
import org.junit.jupiter.api.Test;
import org.junit.jupiter.api.BeforeAll;
import org.junit.jupiter.api.BeforeEach;
import org.junit.jupiter.api.AfterAll;
import org.junit.jupiter.api.AfterEach;
import com.codeborne.selenide.Configuration;
import com.codeborne.selenide.Selectors;
import com.codeborne.selenide.Selenide;
import com.codeborne.selenide.WebDriverRunner;
import static com.codeborne.selenide.Condition.*;
import static com.codeborne.selenide.Selenide.*;
import static org.junit.jupiter.api.Assertions.*;
import org.openqa.selenium.Dimension;
import org.openqa.selenium.Keys;
import java.util.*;
public class DefaultSuiteTest {
  private Map<String, Object> vars;
  @BeforeEach
  public void setUp() {
    Configuration.browser = \\"chrome\\";
    vars = new HashMap<>();
  }
  @AfterEach
  public void tearDown() {
  }
  public String waitForWindow(int timeout) {
    Selenide.sleep(timeout);
    Set<String> whNow = WebDriverRunner.getWebDriver().getWindowHandles();
    Set<String> whThen = (Set<String>) vars.get(\\"window_handles\\");
    if (whNow.size() > whThen.size()) {
      whNow.removeAll(whThen);
    }
    return whNow.iterator().next();
  }
  public void selectwindow() {
    Selenide.open(\\"http://the-internet.herokuapp.com/windows\\");
    vars.put(\\"window_handles\\", WebDriverRunner.getWebDriver().getWindowHandles());
    $(Selectors.byLinkText(\\"Click Here\\")).click();
    vars.put(\\"win9425\\", waitForWindow(2000));
    Selenide.switchTo().window(vars.get(\\"win9425\\").toString());
    assertEquals(Selenide.title(), \\"New Window\\");
  }
  @Test
  public void reuse() {
    selectwindow();
  }
}
"
`;

exports[`Code Export Java Selenide should export a suite with just one new window util method when there are multiple commands that open a new window 1`] = `
"// Generated by Selenium IDE
import org.junit.jupiter.api.Test;
import org.junit.jupiter.api.BeforeAll;
import org.junit.jupiter.api.BeforeEach;
import org.junit.jupiter.api.AfterAll;
import org.junit.jupiter.api.AfterEach;
import com.codeborne.selenide.Configuration;
import com.codeborne.selenide.Selectors;
import com.codeborne.selenide.Selenide;
import com.codeborne.selenide.WebDriverRunner;
import static com.codeborne.selenide.Condition.*;
import static com.codeborne.selenide.Selenide.*;
import static org.junit.jupiter.api.Assertions.*;
import org.openqa.selenium.Dimension;
import org.openqa.selenium.Keys;
import java.util.*;
public class DefaultSuiteTest {
  private Map<String, Object> vars;
  @BeforeEach
  public void setUp() {
    Configuration.browser = \\"chrome\\";
    vars = new HashMap<>();
  }
  @AfterEach
  public void tearDown() {
  }
  public String waitForWindow(int timeout) {
    Selenide.sleep(timeout);
    Set<String> whNow = WebDriverRunner.getWebDriver().getWindowHandles();
    Set<String> whThen = (Set<String>) vars.get(\\"window_handles\\");
    if (whNow.size() > whThen.size()) {
      whNow.removeAll(whThen);
    }
    return whNow.iterator().next();
  }
  public void selectwindow() {
    Selenide.open(\\"http://the-internet.herokuapp.com/windows\\");
    vars.put(\\"window_handles\\", WebDriverRunner.getWebDriver().getWindowHandles());
    $(Selectors.byLinkText(\\"Click Here\\")).click();
    vars.put(\\"win9425\\", waitForWindow(2000));
    Selenide.switchTo().window(vars.get(\\"win9425\\").toString());
    assertEquals(Selenide.title(), \\"New Window\\");
  }
  @Test
  public void reuse() {
    selectwindow();
  }
  @Test
  public void selectwindow() {
    Selenide.open(\\"http://the-internet.herokuapp.com/windows\\");
    vars.put(\\"window_handles\\", WebDriverRunner.getWebDriver().getWindowHandles());
    $(Selectors.byLinkText(\\"Click Here\\")).click();
    vars.put(\\"win9425\\", waitForWindow(2000));
    Selenide.switchTo().window(vars.get(\\"win9425\\").toString());
    assertEquals(Selenide.title(), \\"New Window\\");
  }
}
"
`;

exports[`Code Export Java Selenide should export a test 1`] = `
"// Generated by Selenium IDE
import org.junit.jupiter.api.Test;
import org.junit.jupiter.api.BeforeAll;
import org.junit.jupiter.api.BeforeEach;
import org.junit.jupiter.api.AfterAll;
import org.junit.jupiter.api.AfterEach;
import com.codeborne.selenide.Configuration;
import com.codeborne.selenide.Selectors;
import com.codeborne.selenide.Selenide;
import com.codeborne.selenide.WebDriverRunner;
import static com.codeborne.selenide.Condition.*;
import static com.codeborne.selenide.Selenide.*;
import static org.junit.jupiter.api.Assertions.*;
import org.openqa.selenium.Dimension;
import org.openqa.selenium.Keys;
import java.util.*;
public class LoginTest {
  private Map<String, Object> vars;
  @BeforeEach
  public void setUp() {
    Configuration.browser = \\"chrome\\";
    vars = new HashMap<>();
  }
  @AfterEach
  public void tearDown() {
  }
  @Test
  public void login() {
    Selenide.open(\\"http://the-internet.herokuapp.com/login\\");
    WebDriverRunner.getWebDriver().manage().window().setSize(new Dimension(1440, 1177));
    $(\\"#username\\").val(\\"tomsmith\\");
    $(\\"#password\\").val(\\"SuperSecretPassword!\\");
    $(\\"#login button\\").click();
    $(\\".flash.success\\").shouldHave(text(\\"You logged into a secure area!\\\\\\\\\\\\\\\\n×\\"));
  }
}
"
`;

exports[`Code Export Java Selenide should export a test with a grid configuration 1`] = `
"// Generated by Selenium IDE
import org.junit.jupiter.api.Test;
import org.junit.jupiter.api.BeforeAll;
import org.junit.jupiter.api.BeforeEach;
import org.junit.jupiter.api.AfterAll;
import org.junit.jupiter.api.AfterEach;
import com.codeborne.selenide.Configuration;
import com.codeborne.selenide.Selectors;
import com.codeborne.selenide.Selenide;
import com.codeborne.selenide.WebDriverRunner;
import static com.codeborne.selenide.Condition.*;
import static com.codeborne.selenide.Selenide.*;
import static org.junit.jupiter.api.Assertions.*;
import org.openqa.selenium.Dimension;
import org.openqa.selenium.Keys;
import java.util.*;
public class LoginTest {
  private Map<String, Object> vars;
  @BeforeEach
  public void setUp() {
    Configuration.browser = \\"firefox\\";
    Configuration.remote = \\"http://localhost:4444/wd/hub\\";
    vars = new HashMap<>();
  }
  @AfterEach
  public void tearDown() {
  }
  @Test
  public void login() {
    Selenide.open(\\"http://the-internet.herokuapp.com/login\\");
    WebDriverRunner.getWebDriver().manage().window().setSize(new Dimension(1440, 1177));
    $(\\"#username\\").val(\\"tomsmith\\");
    $(\\"#password\\").val(\\"SuperSecretPassword!\\");
    $(\\"#login button\\").click();
    $(\\".flash.success\\").shouldHave(text(\\"You logged into a secure area!\\\\\\\\\\\\\\\\n×\\"));
  }
}
"
`;

exports[`Code Export Java Selenide should export a test with a reused test method 1`] = `
"// Generated by Selenium IDE
import org.junit.jupiter.api.Test;
import org.junit.jupiter.api.BeforeAll;
import org.junit.jupiter.api.BeforeEach;
import org.junit.jupiter.api.AfterAll;
import org.junit.jupiter.api.AfterEach;
import com.codeborne.selenide.Configuration;
import com.codeborne.selenide.Selectors;
import com.codeborne.selenide.Selenide;
import com.codeborne.selenide.WebDriverRunner;
import static com.codeborne.selenide.Condition.*;
import static com.codeborne.selenide.Selenide.*;
import static org.junit.jupiter.api.Assertions.*;
import org.openqa.selenium.Dimension;
import org.openqa.selenium.Keys;
import java.util.*;
public class ValidcredentialsTest {
  private Map<String, Object> vars;
  @BeforeEach
  public void setUp() {
    Configuration.browser = \\"chrome\\";
    vars = new HashMap<>();
  }
  @AfterEach
  public void tearDown() {
  }
  public void login() {
    Selenide.open(\\"https://the-internet.herokuapp.com/\\");
    $(Selectors.byLinkText(\\"Form Authentication\\")).click();
    $(\\"#username\\").val(vars.get(\\"username\\").toString());
    $(\\"#password\\").val(vars.get(\\"password\\").toString());
    $(\\".fa\\").click();
  }
  @Test
  public void validcredentials() {
    vars.put(\\"username\\", \\"tomsmith\\");
    vars.put(\\"password\\", \\"SuperSecretPassword!\\");
    login();
  }
}
"
`;

exports[`Code Export Java Selenide should export a test with commands that open a new window 1`] = `
"// Generated by Selenium IDE
import org.junit.jupiter.api.Test;
import org.junit.jupiter.api.BeforeAll;
import org.junit.jupiter.api.BeforeEach;
import org.junit.jupiter.api.AfterAll;
import org.junit.jupiter.api.AfterEach;
import com.codeborne.selenide.Configuration;
import com.codeborne.selenide.Selectors;
import com.codeborne.selenide.Selenide;
import com.codeborne.selenide.WebDriverRunner;
import static com.codeborne.selenide.Condition.*;
import static com.codeborne.selenide.Selenide.*;
import static org.junit.jupiter.api.Assertions.*;
import org.openqa.selenium.Dimension;
import org.openqa.selenium.Keys;
import java.util.*;
public class SelectwindowTest {
  private Map<String, Object> vars;
  @BeforeEach
  public void setUp() {
    Configuration.browser = \\"chrome\\";
    vars = new HashMap<>();
  }
  @AfterEach
  public void tearDown() {
  }
  public String waitForWindow(int timeout) {
    Selenide.sleep(timeout);
    Set<String> whNow = WebDriverRunner.getWebDriver().getWindowHandles();
    Set<String> whThen = (Set<String>) vars.get(\\"window_handles\\");
    if (whNow.size() > whThen.size()) {
      whNow.removeAll(whThen);
    }
    return whNow.iterator().next();
  }
  @Test
  public void selectwindow() {
    Selenide.open(\\"http://the-internet.herokuapp.com/\\");
    WebDriverRunner.getWebDriver().manage().window().setSize(new Dimension(1260, 1027));
    $(Selectors.byLinkText(\\"Multiple Windows\\")).click();
    vars.put(\\"window_handles\\", WebDriverRunner.getWebDriver().getWindowHandles());
    $(Selectors.byLinkText(\\"Click Here\\")).click();
    vars.put(\\"win9625\\", waitForWindow(2000));
    vars.put(\\"root\\", WebDriverRunner.getWebDriver().getWindowHandle());
    Selenide.switchTo().window(vars.get(\\"win9625\\").toString());
    assertEquals(Selenide.title(), \\"New Window\\");
    Selenide.switchTo().window(vars.get(\\"root\\").toString());
    assertEquals(Selenide.title(), \\"The Internet\\");
  }
}
"
`;
